optuna/integration/pytorch_lightning.py

```diff
@@ -69,60 +74,86 @@ def on_fit_start(self, trainer: Trainer, pl_module: "pl.LightningModule") -> Non
                     "optuna.integration.PyTorchLightningPruningCallback"
                     " supports only optuna.storages.RDBStorage in DDP."
                 )
+            # It is necessary to store intermediate values directly in the backend storage because
+            # they are not properly propagated to main process due to cached storage.
+            # TODO(Shinichi) Remove intermediate_values from system_attr after PR #4431 is merged.
+            self._trial.storage.set_trial_system_attr(
+                self._trial._trial_id,
+                _INTERMEDIATE_VALUE,
+                dict(),
+            )
```

Comment:
This setup only should be done in the rank zero process? In the following scenario, the value of the `_INTERMEDIATE_VALUE` is wrong.

```
------------------------> time
rank 0: (A) -> (B)
rank 1:                (A)
```
(A): call `on_fit_start` and init the `_INTERMEDIATE_VALUE`
(B): call `on_validation_end` and report the `_INTERMEDIATE_VALUE`
Actually, thanks to the specification of `on_fit_start` and `on_validation_end`, this problem might not occur, but I think it is safe to call this init only in the rank 0 process for future specification change of Pytorch Lightning.

Reply:
Thank you for your careful review. I added an if-block so that `storage.set_trial_system_attr._INTERMEDIATE_VALUE` is initialized only from global zero.